src/Presentation/ListFacetHtmlBuilder.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseFacetedSearch\Presentation;
+
+use MediaWiki\Html\TemplateParser;
+use MediaWiki\Parser\Sanitizer;
+use ProfessionalWiki\WikibaseFacetedSearch\Application\FacetConfig;
+use ProfessionalWiki\WikibaseFacetedSearch\Application\PropertyConstraints;
+use ProfessionalWiki\WikibaseFacetedSearch\Application\ValueCount;
+use ProfessionalWiki\WikibaseFacetedSearch\Application\ValueCounter;
+
+/**
+ * Renders list facets via the `ListFacet.mustache` template.
+ */
+class ListFacetHtmlBuilder implements FacetHtmlBuilder {
+
+	public function __construct(
+		private readonly TemplateParser $parser,
+		private readonly ValueCounter $valueCounter
+	) {
+	}
+
+	/**
+	 * TODO: add integration smoke test
+	 * TODO: add unit tests for the logic. Likley requires some refactoring, ie making the view model accessible
+	 */
+	public function buildHtml( FacetConfig $config, PropertyConstraints $state ): string {
+		return $this->parser->processTemplate(
+			'ListFacet',
+			[
+				'hasToggle' => true, // TODO: use config allowCombineWithChoice and defaultCombineWith
+				'checkboxes' => $this->buildCheckboxesViewModel( $config, $state ),
```

Comment:
Since `DelegatingFacetHtmlBuilder` works on `FacetType` only, are we going to have a different `FacetType` and `FacetHtmlBuilder` for single value radio buttons? Or is that not really a use case any more?
It seems like only the instance type facet might have a single value, but whether we implement that as another facet is not clear (https://github.com/ProfessionalWiki/WikibaseFacetedSearch/issues/40).

Reply:
I would not account for the instance type selection in the facet system.